### PR TITLE
[DS-64] Card Component

### PR DESCRIPTION
**Pull request by victormath12**
This PR implements the Card component on web and native platforms.

### [Web] Simple Card
image

### CODE REVIEW

File: packages/doc/content/components/card/card.mdx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+title: 'Simple'
```
[the remark this answers is not in the document]
Comment (victormath12) [reply packages/doc/content/components/card/card.mdx]: In my opinion, will be a little bit repetitive:

image

[the remark this answers is not in the document]
Comment (victormath12) [reply packages/doc/content/components/card/card.mdx]: What do you think?

[the remark this answers is not in the document]
Comment (allyssonsantos) [reply packages/doc/content/components/card/card.mdx]: Sorry, my mistake, I thought that this was the `metaTitle` property.

File: packages/yoga/src/ThemeProvider/themes/BaseTheme.js [hunk 2]
```diff
@@ -254,6 +254,64 @@ const BaseTheme = ({ primary, secondary }) => {
...
+      radii: radii.rounded,
+      elevation: elevations.small,
+      header: {
+        ribbon: {
+          padding: {
+            x: spacing.xsmall,
+            y: spacing.medium,
+          },
+          radii: {
+            topLeft: 16,
```
Comment (ggdaltoso) [inline packages/yoga/src/ThemeProvider/themes/BaseTheme.js]: Is not the radii token enough?

File: packages/common/src/elevate.js [hunk 3]
```diff
@@ -3,11 +3,11 @@ import hexToRgb from './hexToRgb';
 /**
  * Elevation function
  *
- * @param {String} color=#000 - Color to elevate
+ * @param {String} color=#9898A6 - Color to elevate
  * @param {Number} level - Elevation level. It must be between 0 and 4
  * @returns {String}
  */
-function elevate(color = '#000', level) {
+function elevate(color = '#9898A6', level) {
```
[the remark this answers is not in the document]
Comment (victormath12) [reply packages/common/src/elevate.js]: I changed the implementation in this case:

in `packages/common/src/elevate.js` I returned the last default value **#000**

and in `packages/tokens/src/global/elevations.js` I've added the new color as a parameter of elevate() function, as you can see below:

```

const elevations = elevate(colors.gray[3]);

[
  elevations.zero,
  elevations.small,
  elevations.medium,
  elevations.large,
] = elevations;

export default elevations;

```

[the remark this answers is not in the document]
Comment (victormath12) [reply packages/common/src/elevate.js]: what do you think?

File: packages/yoga/src/Card/native/Plan.jsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        <Card.Header ribbon={ribbon} style={{ marginBottom: 40 }}>
+          <Title selected={selected}>{plan.name}</Title>
+        </Card.Header>
+        <Price selected={selected}>{plan.price}</Price>
+        <Period selected={selected}>/{plan.period}</Period>
+        <Card.Footer style={{ marginTop: 20 }}>
+          <GymsQuantity selected={selected}>{plan.gyms} gyms</GymsQuantity>
+        </Card.Footer>
+      </TouchableOpacity>
+    </CardPlan>
```
[the remark this answers is not in the document]
Comment (ggdaltoso) [reply packages/yoga/src/Card/native/Plan.jsx]: You could pass it as `attrs` too:

```
const CardPlan = styled(Card)({ as: TouchableOpacity })
```

File: packages/common/src/elevate.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (allyssonsantos) [reply packages/common/src/elevate.js]: I think that the best way! <3 @victormath12

File: packages/yoga/src/Card/native/Plan.jsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    font-size: ${font.size}px;
+    font-weight: ${font.weight};
+    ${selected ? `color: ${color.selected}` : `color: ${font.color}`}
+  `}
+`;
+
+const Plan = ({ plan, selected, ribbon, onCardPress, ...rest }) => {
+  return (
+    <CardPlan selected={selected} {...rest}>
+      <TouchableOpacity testID="touchable" onPress={onCardPress}>
```
[the remark this answers is not in the document]
Comment (allyssonsantos) [reply packages/yoga/src/Card/native/Plan.jsx]: You can receive it via the `...rest` and in the test file pass it as a prop to the test function.

File: packages/yoga/src/Card/native/Card.jsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    border-radius: ${radii}px;
+    background-color: ${colors.white};
+    box-shadow: ${elevation};
+    elevation: 4;
+  `}
+`;
+
+const Card = ({ children, ...rest }) => (
+  <CardContainer {...rest}>{children}</CardContainer>
+);
```
[the remark this answers is not in the document]
Comment (victormath12) [reply packages/yoga/src/Card/native/Card.jsx]: The Doc doesn't accept styled-component as a documented component, this cause a error in the Table of Props (PropsTable), as you can see below:

image

File: packages/yoga/src/Card/native/Plan.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (victormath12) [reply packages/yoga/src/Card/native/Plan.jsx]: Awasome <3 I will do it

File: packages/yoga/src/Card/native/CardFooter.jsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+const CardFooter = ({ children, ...rest }) => (
+  <CardFooterWrapper {...rest}>{children}</CardFooterWrapper>
+);
+
+CardFooter.propTypes = {
+  children: node,
+};
+
+CardFooter.defaultProps = {
+  children: 'Text',
```
Comment (alan-oliv) [inline packages/yoga/src/Card/native/CardFooter.jsx]: let's use undefined?

File: packages/yoga/src/Card/native/CardHeader.jsx [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+  align-items: center;
+  justify-content: space-between;
+`;
+
+const CardHeaderContent = styled.View`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  ${({ ribbon }) => `
+    ${ribbon ? `flex-basis: 40%;` : `flex-basis: 100%;`}
```
Comment (alan-oliv) [inline packages/yoga/src/Card/native/CardHeader.jsx]: Can you use a ternary operator here, please?

[the remark this answers is not in the document]
Comment (victormath12) [reply packages/yoga/src/Card/native/CardHeader.jsx]: I will stay with this solution in this moment. The Design team are discussing about the title and ribbon in this Plan Card.

File: packages/labnative/pages/Card.jsx [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+const CardPage = () => {
+  return (
+    <ScrollView
+      style={{
+        width: '100%',
+        height: '100%',
+        padding: 10,
+      }}
+    >
+      <StyledText>Card simple</StyledText>
```
Comment (ggdaltoso) [inline packages/labnative/pages/Card.jsx]: "Card simple" or "Simple Card"?

File: packages/yoga/src/Card/native/Card.test.jsx [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+          </Card>
+        </ThemeProvider>,
+      );
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe.skip('Card Plan Tests', () => {
+    it('should match texts with Plan Card', () => {
+      console.log(CardPlan);
```
Comment (ggdaltoso) [inline packages/yoga/src/Card/native/Card.test.jsx]: Please, remove this console

File: packages/yoga/src/Card/native/Plan.jsx [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+import Card from './Card';
+
+const CardPlanWrapper = styled(TouchableOpacity)(
+  ({
+    selected,
+    theme: {
+      components: { card },
+    },
+  }) => `
+  ${selected && `background-color: ${card.plan.background.selected};`}
```
Comment (ggdaltoso) [inline packages/yoga/src/Card/native/Plan.jsx]:
```suggestion
  ${selected ? `background-color: ${card.plan.background.selected};` : ''}
```

File: packages/yoga/src/Card/native/Plan.jsx [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+    name: string,
+    price: string,
+    period: string,
+    gyms: string,
+  }).isRequired,
+  ribbon: shape({
+    label: string,
+    secondary: bool,
+  }),
+  onPress: func,
```
Comment (ggdaltoso) [inline packages/yoga/src/Card/native/Plan.jsx]: You don't need to specify `onPress` function anymore

File: packages/yoga/src/Card/native/Header.jsx [hunk 13]
```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { node, shape, string, bool } from 'prop-types';
+import styled from 'styled-components';
+
+const CardHeaderWrapper = styled.View`
```
Comment (ggdaltoso) [inline packages/yoga/src/Card/native/Header.jsx]: Is this component really necessarily?

File: packages/yoga/src/Card/native/Header.jsx [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+  position: relative;
+  width: 100%;
+  flex-direction: row;
```
Comment (allyssonsantos) [inline packages/yoga/src/Card/native/Header.jsx]: flex-direction defaults value is already `row`.
```suggestion

```

File: packages/yoga/src/Card/native/Header.jsx [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+  align-content: center;
+  align-items: center;
+  justify-content: space-between;
+`;
+
+const HeaderContent = styled.View`
+  flex-direction: row;
```
Comment (allyssonsantos) [inline packages/yoga/src/Card/native/Header.jsx]: flex-direction defaults value is already `row`.
```suggestion

```

File: packages/yoga/src/Card/web/Header.jsx [hunk 16]
```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { node, shape, string, bool } from 'prop-types';
+import styled from 'styled-components';
+
+const CardHeaderWrapper = styled.div`
+  display: flex;
+  width: 100%;
+  flex-direction: row;
```
Comment (allyssonsantos) [inline packages/yoga/src/Card/web/Header.jsx]: flex-direction defaults value is already `row`.
```suggestion

```

File: packages/yoga/src/Card/web/Header.jsx [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+  align-content: center;
+  align-items: center;
+`;
+
+const HeaderContent = styled.div`
+  display: flex;
+  flex-direction: row;
```
Comment (allyssonsantos) [inline packages/yoga/src/Card/web/Header.jsx]: flex-direction defaults value is already `row`.
```suggestion

```

**Pull request merged**